### ISSUE TITLE
[Bug Report] form_pretrained() always downloads the same weights when using checkpoint-labels

### ISSUE THREAD

**Issue by matthiasdellago** [TransformerLensOrg/TransformerLens#453]
**Describe the bug**
form_pretrained() seems to always downloads the same weights when using checkpoint-labels.
checkpoint_index works.

**Code example**

```
import torch
from transformer_lens.loading_from_pretrained import get_checkpoint_labels
from transformer_lens import utils, HookedTransformer, HookedTransformerConfig, FactoredMatrix, ActivationCache

model_name = "solu-1l-pile"
checkpoint_labels, checkpoint_label_type = get_checkpoint_labels(model_name)

first = HookedTransformer.from_pretrained(model_name,
                                  checkpoint_label=checkpoint_labels[0],
                                 ).blocks[0].attn.W_K
last = HookedTransformer.from_pretrained(model_name,
                                  checkpoint_label=checkpoint_labels[10],
                                    ).blocks[0].attn.W_K

print(checkpoint_labels[0], checkpoint_labels[10])
print(torch.equal(first,last))

first_i = HookedTransformer.from_pretrained(model_name,
                                  checkpoint_index=0,
                                 ).blocks[0].attn.W_K
last_i = HookedTransformer.from_pretrained(model_name,
                                  checkpoint_index=10,
                                    ).blocks[0].attn.W_K

print(torch.equal(first_i,last_i))
print(torch.equal(first_i,first))
```

**System Info**
Describe the characteristic of your environment:
 * pip
 * Fedora
 * Python 3.11

**Additional context**
Maybe usage error on my side?
Working around it with indices is easy enough.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

**Comment by jlarson4** [comment]
Hi @matthiasdellago! Sorry it took so long to get you a response on this.

`HookedTransformer.from_pretrained` has no `checkpoint_label` parameter. The valid parameters are `checkpoint_index: Optional[int]` and `checkpoint_value: Optional[int]`. Your use of `checkpoint_label=checkpoint_labels[0]` is silently absorbed by `**from_pretrained_kwargs` and dropped. Since the information is not passed through, both calls loaded the default (latest) checkpoint, producing identical weights.

That being said, I can see where this confusion came from. I am adding a `checkpoint_label` alias in addition to TransformerBridge support for this feature in the next release, a PR will be up shortly tagging this issue
